server/test/auth/auth.spec.js

```diff
@@ -0,0 +1,157 @@
+const chai = require("chai");
+const chaiHttp = require("chai-http");
+const { app } = require("../../app.js");
+const generateToken = require("../../utils/generateToken");
+const db = require("../db");
+
+chai.should();
+chai.use(chaiHttp);
+
+before(async () => {
+  await db.connect();
+});
+after(async () => {
+  await db.clear();
+  await db.close();
+});
+
+describe(`AUTH TEST SUITE`, async () => {
+  const user = await User.create({
+    username: "tolatwo",
+    email: "tola2@gmail.com",
+    password: "password",
+  });
+  const token = generateToken(user._id);
+  describe("/POST /auth/register", () => {
+    it("it should return newly created user", (done) => {
+      const user = {
+        username: "teste",
+        email: "test12@email.com",
+        password: "passwordlongone1",
+      };
```

Comment:
We could move this in the main scope and re-use it for all the test cases below instead of repeating it

Reply:
Actually, if you observe the object carefully you will notice that they are not the same. I created them for different scenarios. After the first one,
```
      const user = {
        username: "teste",
        email: "test12@email.com",
        password: "passwordlongone1",
      };
```
 others were changed to test for scenarios as  listed below.
`"it should return status code 400 for existing email"
     const user = {
        username: "testes",
        email: ~~"test12@email.com"~~,
        password: "passwordlongone1",
      };
 `
`"it should return status code 400 for existing username"`
   const user = {
          username: ~~"teste"~~,
          email: "test121@email.com",
          password: "passwordlongone1",
        };
 `